### PR TITLE
Add a proper logger to fetch

### PR DESCRIPTION
**Pull request by robmorgan**
This PR introduces a proper logger to `fetch` to ensure it's output is in line with other Gruntwork tools. It also makes integration testing much easier and solves the following issues:

 - #30 
 - #89

**Note:** This PR must be merged after #88 as it improves upon it.

#### Design Notes

- I noticed a lot of methods required multiple string params instead of a single `FetchOptions` struct, so I’m directly passing in a logger to follow the same pattern in most methods.
- I added the `logrus` library but only saw the macOS binary increase by 100kb from 8.9M to ~9.0M.

#### Progress

- [x] Replaced all of the `fmt.Printf` calls.
- [x] Introduced the `logrus` library.

### CODE REVIEW

File: checksum.go [hunk 1]
```diff
@@ -20,7 +22,7 @@ func verifyChecksumOfReleaseAsset(assetPath string, checksumMap map[string]bool,
 		keys := reflect.ValueOf(checksumMap).MapKeys()
 		return newError(checksumDoesNotMatch, fmt.Sprintf("Expected to checksum value to be one of %s, but instead got %s for Release Asset at %s. This means that either you are using the wrong checksum value in your call to fetch, (e.g. did you update the version of the module you're installing but not the checksum?) or that someone has replaced the asset with a potentially dangerous one and you should be very careful about proceeding.", keys, computedChecksum, assetPath))
 	}
-	fmt.Printf("Release asset checksum verified for %s\n", assetPath)
+	logger.Printf("Release asset checksum verified for %s\n", assetPath)
```
Comment (brikis98) [inline checksum.go:25]: If we have a proper logger now, we should switch from `Printf` to `Infof`, `Debugf`, `Errorf`, etc, as appropriate.

File: formatter.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
```
Comment (brikis98) [inline formatter.go:11]: The [gruntwork-cli](https://github.com/gruntwork-io/gruntwork-cli) library already has logrus built in and configured. Would it make more sense to use that? Ideally, we'd move all our CLI tools to use this library under the hood. Perhaps using it just for logging here is an incremental first step?

**Review by brikis98 — COMMENTED** [review]
Nice, thanks Rob!

We should probably add a `--log-level` param so you can control the amount of logging.

**Comment by robmorgan** [comment]
> Nice, thanks Rob!
> 
> We should probably add a `--log-level` param so you can control the amount of logging.

Thanks! Yeah, I intentionally avoided that and instead defaulted to 'INFO'. I wasn't sure if it made sense because I would either suppress a lot of the existing messages by default or we'd need to sprinkle in a few debug messages around the place to actually make it useful. I thought it would be easier to leave it out to under complicate this PR, but I agree it's something we'd want in the future. I could add it but default it to 'INFO' then people can override it to 'ERROR' or 'DEBUG'. What do you think?

File: formatter.go (hunk 2 above)
Comment (robmorgan) [reply formatter.go:11]: I looked at Terragrunt and noticed somebody had contributed a PR with `logrus` about a week ago so I thought that might have been the way forward, but I'm happy to use `gruntwork-cli`. I noticed this repo is using a much older version of `https://github.com/urfave/cli`, so I was a bit worried about compatibility issues. Should I look into swapping it over?

Comment (robmorgan) [reply formatter.go:11]: Yeah I just realized I could only use `gruntwork-cli` for logging as a first step like you mentioned then move over to the other collections at a later stage. I'll port this over.

File: checksum.go (hunk 1 above)
Comment (robmorgan) [reply checksum.go:25]: @brikis98 Good point, then I vote `Infof` for most of these calls as users will have come to expect them. Thoughts?

Comment (brikis98) [reply checksum.go:25]: Use your judgment!

File: formatter.go (hunk 2 above)
Comment (brikis98) [reply formatter.go:11]: Roger!

File: checksum.go (hunk 1 above)
Comment (robmorgan) [reply checksum.go:25]: Moving to `Infof`

File: formatter.go (hunk 2 above)
Comment (robmorgan) [reply formatter.go:11]: Ok, refactored to use `gruntwork-cli` instead for logging.

**Comment by robmorgan** [comment]
@brikis98 okay I've added a `--log-level` option and ported the logging code over to `gruntwork-cli`. To actually make the `--log-level` option worthwhile I've sprinkled in a few `DEBUG` log entries, so execute `fetch` using `--log-level debug` to see them.

**Review by brikis98 — APPROVED** [review]
Very nice, thanks Rob! 👍

**Pull request merged**